<a href="https://colab.research.google.com/github/ameesha543/Statistical-Learning-e23095/blob/main/Assignment_2/Question-6.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### 1. Shape Parameters $\alpha$ and $\beta$
Given;<br>
$a=8$<br> $b=20$<br>$c=11$:

$$\alpha = 1 + 4\left( \frac{11-8}{20-8} \right) = 1 + 4\left( \frac{3}{12} \right) = 1 + 1 = 2$$

$$\beta = 1 + 4\left( \frac{20-11}{20-8} \right) = 1 + 4\left( \frac{9}{12} \right) = 1 + 3 = 4$$

### 2. Support of $X$
interval of possible values for the task duration:
$$S_X = [a, b] = [8, 20]$$

### 3. Probability Density Function $f_X(x)$
Substituting $\alpha=2, \beta=4, a=8, b=12$:
$$f(x) = \frac{(x-8)^{2-1}(20-x)^{4-1}}{B(2,4)(20-8)^{2+4-1}} = \frac{(x-8)(20-x)^3}{B(2,4) \cdot 12^5}$$
for $8 \leq x \leq 20$.

### 4. & 5. Probability Space and Random Variable
*   $\Omega$: $[8, 20]$.
*   $\mathfrak F$: Borel $\sigma$-algebra $\mathcal{B}([8, 20])$.
*   $P$: Defined by the integral of $f_X(x)$ over any subset in $\mathfrak F$.
*   $X$: The identity function $X(\omega) = \omega$ mapping outcomes to the number of days.

In [ ]:
import numpy as np

a, b, c = 8, 20, 11

# 6. Expectation
expected_value = (a + 4*c + b) / 6

# 7. Variance
variance = ((expected_value - a) * (b - expected_value)) / 7

# 8. Standard Deviation
std_dev = np.sqrt(variance)

print(f"Expected Time (E[X]): {expected_value:.2f} days")
print(f"Variance (Var(X)): {variance:.2f}")
print(f"Standard Deviation (sigma): {std_dev:.2f} days")

Expected Time (E[X]): 12.00 days
Variance (Var(X)): 4.57
Standard Deviation (sigma): 2.14 days


### 9. PERT vs. Triangular Distribution
*   Triangular Mean: $(8+11+20)/3 = 13.0$ days.
*   PERT Mean: $(8 + 4(11) + 20)/6 = 72/6 = 12.0$ days.

PERT is often preferred because it assigns less weight to the extreme values (the tails) compared to the Triangular distribution, which has a huge linear slope. In this case, the PERT estimate is more conservative towards the 'most likely' value (11) than the Triangular mean (13), effectively smoothing the distribution.

### 10. Practical Interpretation
*   **$a=8$:** Optimistic time (best-case scenario)
*   **$b=20$:** Pessimistic time (worst-case scenario)
*   **$c=11$:** Most likely duration (mode of the distribution)
*   **$E[X] = 12.00$:** The weighted average completion time, used for project scheduling and path analysis
*   **$\sigma \approx 2.14$:** The risk/uncertainty factor; approximately 68% of the time, the task is expected to finish between 9.86 and 14.14 days